master/buildbot/process/build.py

```diff
@@ -190,25 +194,23 @@ def getWorkerName(self):
         return self.workerforbuilder.worker.workername
     deprecatedWorkerClassMethod(locals(), getWorkerName)
 
-    def setupProperties(self):
```

Comment:
This was moved to a static method so that we don't need an instance of a Build to know its properties. The BuildRequestDistributor needs the final build properties in order to figure out whether a certain build request will satisfy the lock constraits before creating a build for it.